Improving designer

Bournvita1998 opened the pull request:
### Description

#15163 
Signed-off-by: Bournvita1998 <mohit.kuri@research.iiit.ac.in>

Refactored 2 JS files of Designer 

Fixes #

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

Bournvita1998 committed:
refactored 2 files as of now

Bournvita1998 committed:
refactored 2 files

Signed-off-by: Bournvita1998 <mohit.kuri@research.iiit.ac.in>

williamdes commented:
@Bournvita1998 I see that you have made great improvements, can you fix https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/517394475#L645 before I review the PR please ?

mauriciofauth commented on js/designer/database.js:
```diff
@@ -2,8 +2,7 @@ var designer_tables = [{ name: 'pdf_pages', key: 'pg_nr', auto_inc: true },
 var DesignerOfflineDB = (function () {
-    var designerDB = {};
-    var datastore = null;
+    var designerDB = {}, datastore = null;
```
You should split `var` declarations into multiple statements.

Bournvita1998 replied:
I remembered that we should not declare the variables in multiple lines as per good coding practices. But found that it wrong :(. Thus reverted the changes. Also due to the same reason its showing linting error (which I have removed rn, will be displayed in the new commit).

mauriciofauth commented on js/designer/history.js:
```diff
@@ -170,43 +162,50 @@ function history_edit (index) {
-        document.getElementById('query_where').style.zIndex = '103';
-        document.getElementById('query_where').style.visibility = 'visible';
-        document.getElementById('query_where').style.display = 'block';
+        $("#eQuery").value = history_array[index].get_obj().getquery(); // get_obj(),get_query might be removed
```
Why replace `document.getElementById` with the jQuery ID selector?

Bournvita1998 replied:
I was trying to convert some of the JS into jQuery while refactoring the code. Should I continue doing this in this PR itself or should I create a new one?

mauriciofauth replied:
What I want to know is, for example in the `#eQuery` case, what are the advantages in using jQuery instead of using plain JavaScript?

Bournvita1998 replied:
Ohh!!, So do you want to compare JS vs jQuery. Right?

Bournvita1998 replied:
@mauriciofauth I would like to suggest that if you hadn't looked my proposal on "Refactoring and Improving Designer" till now, please have a look at that you will definitely find the answer for "what are the advantages in using jQuery instead of using plain JavaScript?"

Bournvita1998 replied:
@williamdes Here is the link to the proposal :p 
https://docs.google.com/document/d/16rg0DImKFuW_dwxGXbgncJQ0NjZuglXVkRF2rUPzrwA/edit?usp=sharing

mauriciofauth commented:
Hello. Thanks for your contribution!

We require a full name for legal compliance reasons. Could you use your full name in the signature tag?
For example: `Signed-off-by: Jane Developer <jane@example.org>`

See [Developer's Certificate of Origin](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39)

Bournvita1998 committed:
Removed the linting errors
Signed-off-by: Bournvita1998 <mohit.kuri@research.iiit.ac.in>

Bournvita1998 committed:
Improvements in move.js
Signed-off-by: Bournvita1998 <mohit.kuri@research.iiit.ac.in>

Bournvita1998 commented:
@williamdes If everything is fine, should I go ahead with other changes?

williamdes commented:
@Bournvita1998 I would say yes, @mauriciofauth do you agree ?

Bournvita1998 committed:
fiexs #15163

Signed-off-by: Bournvita1998 <mohit.kuri@research.iiit.ac.in>

Improved the performance of all the JS files of the designer and imrpoved the names of the new variables used.

Bournvita1998 commented:
@williamdes I am done with all the changes, please review.

Bournvita1998 commented:
@williamdes @mauriciofauth I am thinking of closing this PR and create new PRs in a step by step manner as mentioned in my proposal. Its because I will be working on the designer project this summer. Should I go ahead?
Also, can you please review my proposal and suggest any changes if required?
https://docs.google.com/document/d/16rg0DImKFuW_dwxGXbgncJQ0NjZuglXVkRF2rUPzrwA/edit?usp=sharing

williamdes commented:
@Bournvita1998 I agree, good proposal !
